### PR TITLE
[SPARK-22348][SQL] The table cache providing ColumnarBatch should also do partition batch pruning

### PR DESCRIPTION
**Pull request by viirya**
## What changes were proposed in this pull request?

We enable table cache `InMemoryTableScanExec` to provide `ColumnarBatch` now. But the cached batches are retrieved without pruning. In this case, we still need to do partition batch pruning.

## How was this patch tested?

Existing tests.

### CODE REVIEW

**Comment by kiszk** [comment]
Good catch, thank you. Would it be possible to add a test case for pruning with table cache?

**Comment by viirya** [comment]
@kiszk Thanks. I've roughly checked existing tests. Seems that there are related ones for pruning the table cache. Let me see if I can add one.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/InMemoryTableScanExec.scala [hunk 1]
```diff
@@ -201,35 +193,50 @@ case class InMemoryTableScanExec(
...
+
+    if (enableAccumulators) {
+      readPartitions.setValue(0)
+      readBatches.setValue(0)
+    }
+
+    // Using these variables here to avoid serialization of entire objects (if referenced directly)
+    // within the map Partitions closure.
+    val relOutput: AttributeSeq = relation.output
+    val buffers = filteredCachedBatches()
```
Comment (CodingCat) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/InMemoryTableScanExec.scala]: this seems to be a leftover of copy-paste

Comment (viirya) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/InMemoryTableScanExec.scala]: Removed. Thanks.

**Comment by viirya** [comment]
also cc @cloud-fan for review.

**Comment by cloud-fan** [comment]
good catch! merging to master, thanks!

**Pull request closed without merging**